packages/synapse-react-client/src/components/Authentication/LoginMethodButton.tsx

```diff
@@ -0,0 +1,72 @@
+import { Box, Button } from '@mui/material'
+import React from 'react'
+import IconSvg, { IconName } from '../IconSvg'
+import LastLoginInfo, { getLastLoginInfo } from './LastLoginInfo'
+import { LoginMethod, getLoginMethodFriendlyName } from './LoginMethod'
+
+export type LoginMethodButtonProps = {
+  loginMethod: LoginMethod
+  iconName: IconName
+  onClick: React.MouseEventHandler<HTMLButtonElement>
+}
+
+export default function LoginMethodButton(props: LoginMethodButtonProps) {
+  const { loginMethod, iconName, onClick } = props
+
+  const { lastLoginMethod, ...rest } = getLastLoginInfo()
+  const lastLoginInfo = LastLoginInfo({
+    lastLoginMethod: lastLoginMethod,
+    display: 'sentence',
+    ...rest,
+  })
+
+  const shouldShowLastLoginInfo =
+    lastLoginMethod === loginMethod && lastLoginInfo
```

Comment:
Only show the text if the button's login method matches the previous login method used and the LastLoginInfo component has something to display